rust/lance-arrow/src/bfloat16.rs

```diff
@@ -275,8 +276,13 @@ mod from_arrow {
 impl FloatArray<BFloat16Type> for BFloat16Array {
     type FloatType = BFloat16Type;
 
-    fn as_slice(&self) -> &[<BFloat16Type as ArrowFloatType>::Native] {
-        unsafe { std::mem::transmute(self.inner.value_data()) }
+    fn as_slice(&self) -> &[bf16] {
+        unsafe {
+            slice::from_raw_parts(
+                self.inner.value_data().as_ptr() as *const bf16,
+                self.inner.value_data().len() / 2,
```

Comment:
why divide by two?

Reply:
the inner array is a `binaryArray` which is `[u8]` takes 1 byte per value, it converts to `[bf16]` which takes 2 bytes per value.